misc/dbt-materialize/dbt/include/materialize/macros/adapters.sql

```diff
@@ -96,11 +96,17 @@
         d.name as database,
         s.name as schema,
         o.name,
-        case when o.type = 'materialized-view' then 'materializedview' else o.type end as type
+        case when o.type = 'materialized-view' then 'materializedview'
+             else o.type
+        end as type
     from mz_objects o
+    left join mz_sources so on o.id = so.id
     join mz_schemas s on o.schema_id = s.id and s.name = '{{ schema_relation.schema }}'
     join mz_databases d on s.database_id = d.id and d.name = '{{ schema_relation.database }}'
-    where type in ('table', 'source', 'view', 'materialized-view', 'index', 'sink')
+    where o.type in ('table', 'source', 'view', 'materialized-view', 'index', 'sink')
+      --Exclude subsources and progress subsources, which aren't relevant in this
+      --context and can bork the adapter (see #20483)
+      and coalesce(so.type, '') not in ('subsource', 'progress')
```

Comment:
strictly speaking no `coalesce` is needed here, `so.type not in ...` will work just as well.

Reply:
I was surprised that...it doesn't.

Reply:
Yes you are right, `NULL NOT IN ('something')` is NULL, that is `FALSE` , and you want it to be `TRUE` in order to match indexes and materialized views that have no type. My bad.